### PR TITLE
feat: universal sigver

### PR DESCRIPTION
**Pull request by nksazonov**
To be REBASED on MASTER and merged AFTER [ref/extract-smart-wallet](https://github.com/layer-3/clearsync/tree/ref/extract-smart-wallet)

### CODE REVIEW

File: pkg/universal_sigver/client_test.go [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+)
+
+// FIXME: rename `local_blockchain` to something better
+func TestVerify(t *testing.T) {
+	ctx := context.Background()
+
+	// 1. Setup node, contracts, signer
+	node := local_blockchain.NewEthNode(ctx, t)
+	contracts := local_blockchain.SetupContracts(ctx, t, node)
+	eoa, err := local_blockchain.NewAccountWithBalance(ctx, big.NewInt(42), node)
```
Comment (nksazonov) [inline pkg/universal_sigver/client_test.go:27]: @dpatsora , I was referring to these functions. `local_blockchain` package name does not seem to provide any context of what these functions do.
I think, we should rename it to something like `aa-test-infra`.
@maxpushka, @dpatsora  , what do you think?

Comment (nksazonov) [reply pkg/universal_sigver/client_test.go:27]: Also, you can see that other functions (namely, `NewBundler` and `SendNative`) are being used in this test.

File: pkg/universal_sigver/client.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	PackERC6492Sig(ctx context.Context, ownerAddress common.Address, index decimal.Decimal, sig []byte) ([]byte, error)
+}
+
+type backend struct {
+	provider          *ethclient.Client
+	smartWalletConfig *smart_wallet.Config
+	entryPointAddress *common.Address
+}
+
+func NewUniversalSigver(provider *ethclient.Client, smartWalletConfig *smart_wallet.Config, entryPointAddress *common.Address) Client {
```
Comment (dpatsora) [inline pkg/universal_sigver/client.go:29]: I would prefer `NewUniversalSig**V**er`

File: pkg/universal_sigver/client_test.go (hunk 1 above)
Comment (dpatsora) [reply pkg/universal_sigver/client_test.go:27]: aa-test-infra +1

**Review by dpatsora — APPROVED** [review]

File: pkg/universal_sigver/client_test.go (hunk 1 above)
Comment (maxpushka) [reply pkg/universal_sigver/client_test.go:27]: I vote for `infra/local` and `infra/public` to better align with Go code style recommendations.
(my longer comment on the topic is [here](https://github.com/layer-3/clearsync/pull/224#discussion_r1528335762))

**Pull request closed without merging**